src/test/java/com/uid2/core/vertx/TestSitesMetadataPath.java

```diff
@@ -0,0 +1,130 @@
+
+    @Mock
+    private OperatorJWTTokenProvider operatorJWTTokenProvider;
+    @Mock
+    private JwtService jwtService;
+
+    // we need trusted to skip the attestation procedure or otherwise the core encpoint call made in this file will
+    // fail at the attestation handler
+    private static final String attestationProtocol = "trusted";
+
+    @BeforeEach
+    void deployVerticle(Vertx vertx, VertxTestContext testContext) throws Throwable {
+        attestationService = new AttestationService();
+        SecretStore.Global.load(((JsonObject) Json.decodeValue(openFile("/com.uid2.core/testSiteSpecificMetadata/test-secrets.json"))));
+        ConfigStore.Global.load(((JsonObject) Json.decodeValue(openFile("/com.uid2.core/testSiteSpecificMetadata/test-configs-provide-private-site-data.json"))));
+        MockitoAnnotations.initMocks(this);
+        CoreVerticle verticle = new CoreVerticle(cloudStorage, authProvider, attestationService, attestationTokenService, enclaveIdentifierProvider, operatorJWTTokenProvider, jwtService);
+        vertx.deployVerticle(verticle, testContext.succeeding(id -> testContext.completeNow()));
+    }
+
+    private String getUrlForEndpoint(String endpoint) {
+        return String.format("http://127.0.0.1:%d/%s", Const.Port.ServicePortForCore, endpoint);
+    }
+
+    private void fakeAuth(OperatorType operatorType, int siteId) {
+        OperatorKey clientKey = new OperatorKey("test-key", "", "", attestationProtocol, 0, false, siteId, new HashSet<>(), operatorType);
+        when(authProvider.get(any())).thenReturn(clientKey);
+    }
+
+    private void get(Vertx vertx, String endpoint, String body, Handler<AsyncResult<HttpResponse<Buffer>>> handler) {
+        WebClient client = WebClient.create(vertx);
+        client.getAbs(getUrlForEndpoint(endpoint)).sendBuffer(Buffer.buffer(body), handler);
+    }
+
+    @Test
+    void publicOperatorGetsGlobalSites(Vertx vertx, VertxTestContext testContext) throws CloudStorageException, IOException {
+        String metadata = "/com.uid2.core/testSiteSpecificMetadata/sites/metadata.json";
+        String metadataContent = openFile(metadata);
+        String location = ((JsonObject) Json.decodeValue(metadataContent)).getJsonObject("sites").getString("location");
+        when(cloudStorage.download(eq(metadata))).thenReturn(new StringBufferInputStream(metadataContent));
+        when(cloudStorage.preSignUrl(any())).thenAnswer(i -> new URL(i.getArgument(0)));
+        fakeAuth(OperatorType.PUBLIC, 99);
+        get(vertx, "/sites/refresh", "", ar -> {
+            assertTrue(ar.succeeded());
+            HttpResponse response = ar.result();
+            assertEquals(200, response.statusCode());
+            JsonObject json = response.bodyAsJsonObject();
+            String resultLocation = json.getJsonObject("sites").getString("location");
+            assertEquals(resultLocation, location);
+            testContext.completeNow();
+        });
```

Comment:
[Asserting in async code](https://vertx.io/docs/vertx-junit5/java/#_use_any_assertion_library) requires `verify`, `succeeding`, or `failing`.

```suggestion
        get(vertx, "/sites/refresh", "", testContext.succeeding(response -> testContext.verify(() -> {
            assertEquals(200, response.statusCode());
            JsonObject json = response.bodyAsJsonObject();
            String resultLocation = json.getJsonObject("sites").getString("location");
            assertEquals(resultLocation, location);
            testContext.completeNow();
        })));
```